http2/hpack/hpack_test.go

```diff
@@ -72,45 +73,130 @@ func TestDynamicTableSizeEvict(t *testing.T) {
-
-		// C.2.3 Literal Header Field never Indexed
-		// https://httpwg.org/specs/rfc7541.html#rfc.section.C.2.3
-		{"C.2.3", dehex("1008 7061 7373 776f 7264 0673 6563 7265 74"),
-			[]HeaderField{{"password", "secret", true}},
-			[]HeaderField{}},
+		{"broken second request last byte- - grpc timeout", [][]byte{
+			{0x83, 0x86, 0x45, 0x95, 0x62, 0x72, 0xd1, 0x41, 0xfc, 0x1e, 0xca, 0x24, 0x5f, 0x15, 0x85, 0x2a, 0x4b, 0x63, 0x1b, 0x87, 0xeb, 0x19, 0x68, 0xa0, 0xff, 0x41, 0x8b, 0xa0, 0xe4, 0x1d, 0x13, 0x9d, 0x9, 0xb8, 0xd3, 0x6f, 0xb6, 0xcf, 0x5f, 0x8b, 0x1d, 0x75, 0xd0, 0x62, 0xd, 0x26, 0x3d, 0x4c, 0x4d, 0x65, 0x64, 0x7a, 0x8a, 0x9a, 0xca, 0xc8, 0xb4, 0xc7, 0x60, 0x2b, 0xb4, 0xea, 0xe0, 0x40, 0x2, 0x74, 0x65, 0x86, 0x4d, 0x83, 0x35, 0x5, 0xb1, 0x1f, 0x40, 0x89, 0x9a, 0xca, 0xc8, 0xb2, 0x4d, 0x49, 0x4f, 0x6a, 0x7f, 0x86, 0x7d, 0xf7, 0xdf, 0x8, 0x9e, 0xb7},
+			{0x83, 0x86, 0x45, 0x9b, 0x62, 0xbb, 0xe, 0x93, 0xc7, 0x6c, 0xab, 0xee, 0x4d, 0x85, 0x1d, 0x3b, 0x8b, 0x67, 0x73, 0x10, 0xac, 0x61, 0x2a, 0x49, 0x3e, 0xd2, 0xa2, 0xb3, 0xd4, 0x82, 0xff, 0xc3, 0xc2, 0xc1, 0xc0, 0x7f, 0x0, 0x86, 0x7d, 0xf7, 0xde, 0x71},
+		},
+			[][]HeaderField{
+				{
+					pair(":method", "POST"),
+					pair(":scheme", "http"),
+					pair(":path", "/helloworld.Greeter/SayHello"),
+					pair(":authority", "localhost:45953"),
+					pair("content-type", "application/grpc"),
+					pair("user-agent", "grpc-go/1.47.0"),
+					pair("te", "trailers"),
+					pair("grpc-timeout", "9999128u"),
+				},
+				{
+					pair(":method", "POST"),
+					pair(":scheme", "http"),
+					pair(":path", "/protobuf.StreamService/FetchResponse"),
+					pair(":authority", "localhost:45953"),
+					pair("content-type", "application/grpc"),
+					pair("user-agent", "grpc-go/1.47.0"),
+					pair("te", "trailers"),
+					pair("grpc-timeout", "9998638u"),
+				}},
+			[]HeaderField{
+				pair(":path", "/protobuf.StreamService/FetchResponse"),
+				pair("grpc-timeout", "9999128u"),
+				pair("te", "trailers"),
+				pair("user-agent", "grpc-go/1.47.0"),
+				pair("content-type", "application/grpc"),
+				pair(":authority", "localhost:45953"),
+				pair(":path", "/helloworld.Greeter/SayHello")},
+			[]error{nil, DecodingError{errors.New("truncated headers")}},
+		},
 
-		// C.2.4 Indexed Header Field
-		// https://httpwg.org/specs/rfc7541.html#rfc.section.C.2.4
-		{"C.2.4", []byte("\x82"),
-			[]HeaderField{pair(":method", "GET")},
-			[]HeaderField{}},
+		{"minimum dynamic key", [][]byte{
+			{0x83, 0x86, 0x45, 0x95, 0x62, 0x72, 0xd1, 0x41, 0xfc, 0x1e, 0xca, 0x24, 0x5f, 0x15, 0x85, 0x2a, 0x4b, 0x63, 0x1b, 0x87, 0xeb, 0x19, 0x68, 0xa0, 0xff, 0x41, 0x8b, 0xa0, 0xe4, 0x1d, 0x13, 0x9d, 0x9, 0xb8, 0xd3, 0x6f, 0xb6, 0xcf, 0x5f, 0x8b, 0x1d, 0x75},
+			{0x83, 0x86, 0x45, 0x9b, 0x62, 0xbb, 0xe, 0x93, 0xc7, 0x6c, 0xab, 0xee, 0x4d, 0x85, 0x1d, 0x3b, 0x8b, 0x67, 0x73, 0x10, 0xac, 0x61, 0x2a, 0x49, 0x3e, 0xd2, 0xa2, 0xb3, 0xd4, 0x82, 0xff, 0xc3, 0xc2, 0xc1, 0xc0, 0x7f, 0x0, 0x86, 0x7d, 0xf7, 0xde, 0x71},
```

Comment:
what did you change here?

Reply:
left only the part which is related to “:path”, “:authority”, “:method”, “:scheme” which are the interesting parts of the dynamic table, and then showed that still the dynamic table an be created with only those values, even if some part of the request did not captured